cli/smartcontract/smart_contract.go

```diff
@@ -657,12 +657,12 @@ func invokeWithArgs(ctx *cli.Context, acc *wallet.Account, wall *wallet.Wallet,
 	if err != nil {
 		return sender, cli.NewExitError(err, 1)
 	}
-	if signAndPush && resp.State != "HALT" {
+	if resp.State != "HALT" {
 		errText := fmt.Sprintf("Warning: %s VM state returned from the RPC node: %s\n", resp.State, resp.FaultException)
 		if !ctx.Bool("force") {
-			return sender, cli.NewExitError(errText+". Use --force flag to send the transaction anyway.", 1)
+			return sender, cli.NewExitError(errText+". Use --force flag to send/save the transaction anyway.", 1)
```

Comment:
Print `send` or `save` depending on situation? Users need to know exactly what's going to happen with the transaction.